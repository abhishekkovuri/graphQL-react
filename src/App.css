.list {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100vh;
}
header {
  font-size: 36px;
  font-weight: 800;
}

.item {
  border: 1px solid #999;
  border-radius: 10px;
  margin: 10px;
  padding: 10px;
  width: 250px;
  text-align: center;
  font-size: 18px;
  font-weight: 800;
}