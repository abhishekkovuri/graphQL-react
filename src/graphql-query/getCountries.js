import { gql } from "@apollo/client"

export const GET_COUNTRIES = gql`
	query GetCountries {
		getCountries {
			countries {
				countryName
				capital
			}
		}
	}
`
