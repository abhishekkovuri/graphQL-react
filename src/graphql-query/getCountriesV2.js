import { gql } from "@apollo/client"

export const GET_COUNTRIES_V2 = gql`
	query GetCountriesV2 {
		getCountriesV2 {
			countries {
				countryName
				capital
			}
		}
	}
`
