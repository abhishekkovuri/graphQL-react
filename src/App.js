import React from 'react';
import { useLazyQuery, useQuery } from '@apollo/client';
import { GET_COUNTRIES } from './graphql-query/getCountries.js';
import { GET_COUNTRIES_V2 } from "./graphql-query/getCountriesV2"
import './App.css';

const App = () => {
	const { data } = useQuery(GET_COUNTRIES);
	const countries = data?.getCountries?.countries;

	const { data: countriesData } = useQuery(GET_COUNTRIES_V2);
	const anotherCountryList = countriesData?.getCountries?.countries;

	const [lazyLoadBtn, { data: duplicateCountriesData }] =
		useLazyQuery(GET_COUNTRIES);
	const duplicateCountries = duplicateCountriesData?.getCountries?.countries;

	return (
		<section className="list">
			<header> Render Country List </header>
			<section>
				<h1>Original data</h1>
				{countries &&
					countries.map((country) => (
						<div className="item" key={country.countryName}>
							{country.countryName} - {country.capital}
						</div>
					))}
			</section>
			<button onClick={lazyLoadBtn}>Click me</button>
			<section>
				<h1>Duplicate data</h1>
				{duplicateCountries &&
					duplicateCountries.map((country) => (
						<div className="item" key={country.countryName}>
							{country.countryName} - {country.capital}
						</div>
					))}
			</section>
			<section>
				<h1>Another set of countries</h1>
				{anotherCountryList &&
					anotherCountryList.map((country) => (
						<div className="item" key={country.countryName}>
							{country.countryName} - {country.capital}
						</div>
					))}
			</section>
		</section>
	);
};

export default App;
