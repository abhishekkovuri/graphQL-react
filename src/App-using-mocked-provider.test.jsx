import {
	render,
	waitFor,
	fireEvent,
	getAllByText,
} from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import App from './App';
import { GET_COUNTRIES } from './graphql-query/getCountries';
import { GET_COUNTRIES_V2 } from './graphql-query/getCountriesV2';

describe('App test', () => {
	const mocks = [
		{
			request: {
				query: GET_COUNTRIES,
			},
			result: {
				data: {
					getCountries: {
						countries: [
							{
								countryName: 'India',
								capital: 'New Delhi',
							},
						],
					},
				},
			},
		},
		{
			request: {
				query: GET_COUNTRIES_V2,
			},
			result: {
				data: {
					getCountriesV2: {
						countries: [
							{
								countryName: 'Japan',
								capital: 'Tokyo',
							},
						],
					},
				},
			},
		},
	];

	test('should generate snapshot', async () => {
		const { asFragment } = render(
			<MockedProvider mocks={[...mocks]}>
				<App />
			</MockedProvider>
		);

		await waitFor(() => expect(asFragment()).toMatchSnapshot());
	});

	test('should have the countries list generated', async () => {
		const { getByText, queryByText } = render(
			<MockedProvider mocks={[...mocks]}>
				<App />
			</MockedProvider>
		);

		expect(getByText(/Render Country List/i)).toBeInTheDocument();
		await waitFor(() =>
			expect(getByText(/India - New Delhi/i)).toBeInTheDocument()
		);
		await waitFor(() =>
			expect(queryByText(/Japan - Tokyo/i)).not.toBeInTheDocument()
		);
	});

	test('should display duplicate countries list upon click', async () => {
		const { getAllByText, getByText } = render(
			<MockedProvider mocks={[...mocks]}>
				<App />
			</MockedProvider>
		);

		const buttonElement = getByText(/Click me/);
		fireEvent.click(buttonElement);

		await waitFor(() =>
			expect(getAllByText(/India - New Delhi/i).length).toBe(2)
		);
	});
});
