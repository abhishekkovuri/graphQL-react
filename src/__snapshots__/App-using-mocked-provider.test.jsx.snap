// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should generate snapshot 1`] = `
<DocumentFragment>
  <section
    class="list"
  >
    <header>
       Render Country List 
    </header>
    <section>
      <h1>
        Original data
      </h1>
    </section>
    <button>
      Click me
    </button>
    <section>
      <h1>
        Duplicate data
      </h1>
    </section>
    <section>
      <h1>
        Another set of countries
      </h1>
    </section>
  </section>
</DocumentFragment>
`;
